src/App.js

```diff
@@ -2,21 +2,30 @@ import React from "react";
 import TodoList from "./TodoList";
 import AddToDoForm from "./AddToDoForm";
 
-function App() {
-  // const [newTodo, setNewTodo] = React.useState("");
-  const [todoList, setTodoList] = React.useState([]);
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = React.useState(
+    JSON.parse(localStorage.getItem("savedTodoList"))
```

Comment:
Would it be best to handle the case where nothing yet has been stored in local storage? I believe `todolist` would be `null` in that case, which would cause an error when iterating over the list of todos: [here](https://github.com/sadykova/react-to-do-app/blob/be47cff3b5a9ff51ed7adff78574a0ce116cc163/src/TodoList.js#L7).

Reply:
Hello! You mean add an if statement where nothing is in local storage yet to handle null and avoid error?

Reply:
Right, an `if` statement is an option. You could also use `||` to accomplish [setting a default](https://betterprogramming.pub/6-ways-to-set-default-values-in-javascript-b6869fa65cdb):

`let todoList = localStorage.getItem("savedTodoList") || []`

Using the browser console, this is the issue that we want to avoid. The first time, `todoList` is `null`, which causes an error because `null` doesn't have a `map` function. That's what will happen in your code as it's written. An empty array does have a `map` function, so no error.
 
![CleanShot 2022-12-09 at 13 21 37@2x](https://user-images.githubusercontent.com/63003/206768079-5549a251-ec23-463b-8096-aca38dc36251.png)

Reply:
Thank you! Fixed that.